File: jmserver/src/classes/jmserver/services/SecurityService.go
package services
import (
	"../../incryptEncode"
)

type SecurityService struct {
	encoder    incryptEncode.IncryptEncode
}

var __security_service__ *SecurityService;

func GetSecurityServiceInstance() *SecurityService {
	if __security_service__ == nil {
		__security_service__ = &SecurityService{};
		__security_service__.encoder = &incryptEncode.IncryptEncodeFullBinary{};
	}
	return __security_service__;
}

func (x *SecurityService) GetEncoded(a string) string {
	return x.encoder.GetEncoded(a);
}

func (x *SecurityService) GetDecoded(a string) string {
	return x.encoder.GetDecoded(a);
}

File: JobManager/Services/SecurityService.swift
//
//  SecurityService.swift
//  JobMonitor
//
//  Created by Abel Gancsos on 5/9/20.
//  Copyright © 2020 Abel Gancsos. All rights reserved.
//

import Foundation
import IncryptEncode

public final class SecurityService {
    private static var encoder : IncryptEncode = IncryptEncodeFullBinary();
    
    public static func getEncoded(a : String) -> String {
        return encoder.getEncoded(a: a);
    }
    
    public static func getDecoded(a: String) -> String {
        return encoder.getDecoded(a: a);
    }
}
